src/views/ProductDetails.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useState, Fragment } from 'react';
+import { Container, Typography, Grid, Button, makeStyles, LinearProgress } from "@material-ui/core";
+import { RouteComponentProps, useHistory } from "react-router-dom";
+import { findById, deleteById } from '../data/ProductService';
+import AddShoppingCartIcon from '@material-ui/icons/AddShoppingCart';
+import DeleteForeverOutlinedIcon from '@material-ui/icons/DeleteForeverOutlined';
+import { addProductInUserCart } from '../data/CartService';
+import Product from '../domain/Product';
+import { StyledPaper } from '../components/StyledPaper';
+
+
+const useStyles = makeStyles((theme) => ({
+    img: {
+        width: '100%',
+        height: '100%'
+    },
+}));
+
+function ProductDetails(props: RouteComponentProps<any>) {
+    const params = props.match.params
```

Comment:
In loc de asta, poti folosi destructuring pe obiectul de props, in linia 19.
https://medium.com/@lcriswell/destructuring-props-in-react-b1c295005ce0